packages/api-providers/src/polkadot/vanchor-deposit.ts

```diff
@@ -3,31 +3,222 @@
+import { LoggerService } from '@webb-tools/app-util';
+import { ArkworksProvingManager, Note, NoteGenInput, ProvingManagerSetupInput, Utxo } from '@webb-tools/sdk-core';
+import { VAnchorProof } from '@webb-tools/sdk-core/proving/types';
+import { BigNumber } from 'ethers';
+
+import { decodeAddress } from '@polkadot/keyring';
+import { hexToU8a, u8aToHex } from '@polkadot/util';
+import { naclEncrypt, randomAsU8a } from '@polkadot/util-crypto';
+
+import {
+  DepositPayload as IDepositPayload,
+  TransactionState,
+  VAnchorDeposit,
+  VAnchorDepositResults,
+} from '../abstracts';
+import { chainTypeIdToInternalId, computeChainIdType } from '../chains';
+import { WebbError, WebbErrorCodes } from '../webb-error';
+
+const logger = LoggerService.get('PolkadotVBridgeDeposit');
+
+export function currencyToUnitI128(currencyAmount: number) {
+  let bn = BigNumber.from(currencyAmount);
+  return bn.mul(1_000_000_000_000);
+}
+
+// TODO: export this from webb.js
+
+// The Deposit Payload is the note and [treeId]
+type DepositPayload = IDepositPayload<Note, [number]>;
 
-// The Deposit Payload is the note and [treeId, leafHex]
-type DepositPayload = IDepositPayload<Note, [number, string]>;
 /**
  * Webb Anchor API implementation for Polkadot
  **/
+
 export class PolkadotVAnchorDeposit extends VAnchorDeposit<WebbPolkadot, DepositPayload> {
-  generateBridgeNote(
-    anchorId: string | number,
+  private async getleafIndex(leaf: Uint8Array, indexBeforeInsertion: number, treeId: number): Promise<number> {
+    return getLeafIndex(this.inner.api, leaf, indexBeforeInsertion, treeId);
+  }
+
+  async generateBridgeNote(
+    _vanchorId: string | number, // always Zero as there will be only one vanchor
     destination: number,
     amount: number,
     wrappableAssetAddress?: string
   ): Promise<DepositPayload> {
-    throw new Error('Method not implemented.');
+    // Get the currency bridge currency
+    const currency = this.bridgeApi.currency;
+
+    // No currency is selected on the API
+    if (!currency) {
+      logger.error('Not currency/active bridge available');
+      throw new Error('api not ready');
+    }
+    const tokenSymbol = currency.view.symbol;
+    const destChainId = destination;
+    // Chain id of the active API
+    const chainId = await this.inner.api.consts.linkableTreeBn254.chainIdentifier;
+    const chainType = await this.inner.api.consts.linkableTreeBn254.chainType;
+    const sourceChainId = computeChainIdType(Number(chainType.toHex()), Number(chainId));
+    const anchors = await this.bridgeApi.getVariableAnchors();
+    const anchor = anchors[0];
+    const internalChainId = chainTypeIdToInternalId({
+      chainId: Number(chainId),
+      chainType: Number(chainType.toHex()),
+    });
+    // Tree id for the target chain
+    const treeId = anchor.neighbours[internalChainId] as number;
+
+    const noteInput: NoteGenInput = {
+      amount: currencyToUnitI128(amount).toString(),
```

Comment:
This is putting the note amount as `2000000000000` instead of `2`.  The `2` format is used in FixedAnchor, Mixer, and web3 vanchor.

Reply:
On the substrate side, the amount is not just the unit value, the denomination contributes to it as well
so well shall fix this we have a function to calculate that `calcAmount(amount, denomination)`

notes would have just the amount on units

Reply:
I would try to use the `fixed-point-number` in webb.js instead of creating a new function, or use something already out there like ether's `parseUnits`

Reply:
This is the task meant to track this issue: https://github.com/webb-tools/webb-dapp/issues/448